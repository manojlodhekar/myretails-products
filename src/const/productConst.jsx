export default {
    PRODUCT_API_URL: '/assets/data/item-data.json',
    PRODUCT_MAIN_PAGE_SUBHEADER: 'Product Details',
    START_INST: 'Please click button for Products details',
    HEADER_TEXT: 'MyRetails Products Warehouse',
    PRODUCT_HIGHLIGHT: 'Products Highlights',
    FIND_STORE: 'find in store',
    PRO_HELPFUL: 'Most helpful 4-5 star review',    
    CON_HELPFUL: 'Most helpful 1-2 star review',
    ERROR_MESSAGE: 'Oops..Something went Wrong, please try again.',
    TARGET_BIG_LOGO: '/assets/Target-Logo.png',
    TARGET_LOGO: '/assets/Color-Target-logo.png',
    ADD_TO_CART: '/assets/addToCart.png',
    PICK_IN_STORE: '/assets/pickInStore.png',
    SHOPING_BAG_ICON : 'https://payload.cargocollective.com/1/0/6826/4717503/Cart.png',//'/assets/shopping-bag.svg',
    CART_BTN_LABEL: 'ADD TO CART',
    PICK_BTN_LABEL: 'PICK UP IN STORE',
    ADD_TO_REG: 'ADD TO REGISTRY',
    ADD_TO_LIST: 'ADD TO LIST',
    SHARE: 'SHARE',
    MY_RETAILS_INFO: 'MyRetail is a rapidly growing company with HQ in Richmond, VA and over 200 stores across the east coast. The companyâ€™s annual revenue last year was $5 billion and growing at the rate of 10% year over year. myRetail sells general merchandise products, including a fast growing fresh grocery segment. The stores average 80,000 sq. ft. in size and carry around 20,000 products. myRetail wants to provide a multi-channel experience for its customers online.'
}
    