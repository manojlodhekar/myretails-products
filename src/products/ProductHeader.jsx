import React from 'react';
import { Image, } from 'semantic-ui-react'
import productConst from '../const/productConst';

const styles = {
  
    header:{
        paddingLeft: 10,
        textAlign: 'left',
        paddingBottom: 10,
        borderBottom: '1px solid grey'
    },
    headerTxt:{
        paddingLeft: 10,
        textAlign: 'left',
        fontSize: 24,
        fontWeight: 'bold',
        lineHeight: 1.2,
        color: 'white'
    },
    logoCls:{
        maxWidth: 55, 
        float: 'right', 
        textAlign: 'right'
    },
    subHeaderText:{
        paddingLeft: 10,
        textAlign: 'left',
        fontSize: 18,
        color: 'white'
    },
    headMessage:{
        width: '100%'
    },
    cartIcon:{
        backgroundColor: '#fff',
       //margin: '2px 8px 2px 5px'
    },
    cart:{
        
    }

}


const ProductHeader = (props) =>{

    return(
        <div className="headContainer" style={{display: 'flex'}}>
            <div>
                <Image src={productConst.TARGET_LOGO} style={styles.logoCls} />
            </div>
            <div style={styles.headMessage}>
                <div style={styles.headerTxt}>
                        {productConst.HEADER_TEXT}
                </div>
                <div style={styles.subHeaderText}>
                        {props.subHeaderText}
                </div>
            </div>
        </div>
    )
}
export default ProductHeader;