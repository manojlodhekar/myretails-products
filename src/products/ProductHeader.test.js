import React from 'react';
import ReactDOM from 'react-dom';
import ProductHeader from './ProductHeader';
import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { shallow } from 'enzyme';
import productConst from '../const/productConst';

jest.mock('react-dom');

Enzyme.configure({ adapter: new Adapter() });
let productHeader;
beforeEach(()=>{
  productHeader = shallow(<ProductHeader subHeaderText="My retails Product Details" />);

})
describe('ProductHeader is defined', () => {

  it('should render correctly', () => {
    expect(productHeader).toHaveLength(1);
  });

  it('should render inner component', () => {
    expect(productHeader).toMatchSnapshot();
  });
  it('should contains passed message', () => {
    expect(productHeader.contains('My retails Product Details')).toBe(true);
  });
  it('should contains default message', () => {
    expect(productHeader.contains(productConst.HEADER_TEXT)).toBe(true);
  });

 });
