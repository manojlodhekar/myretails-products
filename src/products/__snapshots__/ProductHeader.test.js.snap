// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductHeader is defined should render inner component 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ProductHeader
    subHeaderText="My retails Product Details"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div>
          <Image
                    as="img"
                    src="/assets/Color-Target-logo.png"
                    style={
                              Object {
                                        "float": "right",
                                        "maxWidth": 55,
                                        "textAlign": "right",
                                      }
                    }
                    ui={true}
          />
</div>,
        <div
          style={
                    Object {
                              "width": "100%",
                            }
          }
>
          <div
                    style={
                              Object {
                                        "color": "white",
                                        "fontSize": 24,
                                        "fontWeight": "bold",
                                        "lineHeight": 1.2,
                                        "paddingLeft": 10,
                                        "textAlign": "left",
                                      }
                    }
          >
                    MyRetails Products Warehouse
          </div>
          <div
                    style={
                              Object {
                                        "color": "white",
                                        "fontSize": 18,
                                        "paddingLeft": 10,
                                        "textAlign": "left",
                                      }
                    }
          >
                    My retails Product Details
          </div>
</div>,
      ],
      "className": "headContainer",
      "style": Object {
        "display": "flex",
      },
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <Image
            as="img"
            src="/assets/Color-Target-logo.png"
            style={
                        Object {
                                    "float": "right",
                                    "maxWidth": 55,
                                    "textAlign": "right",
                                  }
            }
            ui={true}
/>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "as": "img",
            "src": "/assets/Color-Target-logo.png",
            "style": Object {
              "float": "right",
              "maxWidth": 55,
              "textAlign": "right",
            },
            "ui": true,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              style={
                            Object {
                                          "color": "white",
                                          "fontSize": 24,
                                          "fontWeight": "bold",
                                          "lineHeight": 1.2,
                                          "paddingLeft": 10,
                                          "textAlign": "left",
                                        }
              }
>
              MyRetails Products Warehouse
</div>,
            <div
              style={
                            Object {
                                          "color": "white",
                                          "fontSize": 18,
                                          "paddingLeft": 10,
                                          "textAlign": "left",
                                        }
              }
>
              My retails Product Details
</div>,
          ],
          "style": Object {
            "width": "100%",
          },
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "MyRetails Products Warehouse",
              "style": Object {
                "color": "white",
                "fontSize": 24,
                "fontWeight": "bold",
                "lineHeight": 1.2,
                "paddingLeft": 10,
                "textAlign": "left",
              },
            },
            "ref": null,
            "rendered": "MyRetails Products Warehouse",
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "My retails Product Details",
              "style": Object {
                "color": "white",
                "fontSize": 18,
                "paddingLeft": 10,
                "textAlign": "left",
              },
            },
            "ref": null,
            "rendered": "My retails Product Details",
            "type": "div",
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div>
            <Image
                        as="img"
                        src="/assets/Color-Target-logo.png"
                        style={
                                    Object {
                                                "float": "right",
                                                "maxWidth": 55,
                                                "textAlign": "right",
                                              }
                        }
                        ui={true}
            />
</div>,
          <div
            style={
                        Object {
                                    "width": "100%",
                                  }
            }
>
            <div
                        style={
                                    Object {
                                                "color": "white",
                                                "fontSize": 24,
                                                "fontWeight": "bold",
                                                "lineHeight": 1.2,
                                                "paddingLeft": 10,
                                                "textAlign": "left",
                                              }
                        }
            >
                        MyRetails Products Warehouse
            </div>
            <div
                        style={
                                    Object {
                                                "color": "white",
                                                "fontSize": 18,
                                                "paddingLeft": 10,
                                                "textAlign": "left",
                                              }
                        }
            >
                        My retails Product Details
            </div>
</div>,
        ],
        "className": "headContainer",
        "style": Object {
          "display": "flex",
        },
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <Image
              as="img"
              src="/assets/Color-Target-logo.png"
              style={
                            Object {
                                          "float": "right",
                                          "maxWidth": 55,
                                          "textAlign": "right",
                                        }
              }
              ui={true}
/>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "as": "img",
              "src": "/assets/Color-Target-logo.png",
              "style": Object {
                "float": "right",
                "maxWidth": 55,
                "textAlign": "right",
              },
              "ui": true,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                style={
                                Object {
                                                "color": "white",
                                                "fontSize": 24,
                                                "fontWeight": "bold",
                                                "lineHeight": 1.2,
                                                "paddingLeft": 10,
                                                "textAlign": "left",
                                              }
                }
>
                MyRetails Products Warehouse
</div>,
              <div
                style={
                                Object {
                                                "color": "white",
                                                "fontSize": 18,
                                                "paddingLeft": 10,
                                                "textAlign": "left",
                                              }
                }
>
                My retails Product Details
</div>,
            ],
            "style": Object {
              "width": "100%",
            },
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "MyRetails Products Warehouse",
                "style": Object {
                  "color": "white",
                  "fontSize": 24,
                  "fontWeight": "bold",
                  "lineHeight": 1.2,
                  "paddingLeft": 10,
                  "textAlign": "left",
                },
              },
              "ref": null,
              "rendered": "MyRetails Products Warehouse",
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "My retails Product Details",
                "style": Object {
                  "color": "white",
                  "fontSize": 18,
                  "paddingLeft": 10,
                  "textAlign": "left",
                },
              },
              "ref": null,
              "rendered": "My retails Product Details",
              "type": "div",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
